rma_analytic/tests/test_rma_analytic.py

```diff
@@ -1,32 +1,41 @@
-# -*- coding: utf-8 -*-
-# © 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
 # License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
 
 from odoo.addons.rma.tests import test_rma
 
 
 class TestRmaAnalytic(test_rma.TestRma):
 
-    def setUp(self):
-        super(TestRmaAnalytic, self).setUp()
-        products2move = [(self.product_1, 3), (self.product_2, 5),
-                         (self.product_3, 2)]
-        self.rma_ana_id = self._create_rma_from_move(
-            products2move, 'supplier', self.env.ref('base.res_partner_1'),
-            dropship=False)
+    @classmethod
+    def setUpClass(cls):
+        super(TestRmaAnalytic, cls).setUpClass()
+        products2move = [
+            (cls.product_1, 3),
+            (cls.product_2, 5),
+            (cls.product_3, 2),
+        ]
+        cls.rma_ana_id = cls._create_rma_from_move(
+            products2move,
+            "supplier",
+            cls.env.ref("base.res_partner_2"),
+            dropship=False,
+        )
 
-    def _prepare_move(self, product, qty, src, dest, picking_in):
-        res = super(TestRmaAnalytic, self)._prepare_move(
+    @classmethod
+    def _prepare_move(cls, product, qty, src, dest, picking_in):
+        res = super(TestRmaAnalytic, cls)._prepare_move(
             product, qty, src, dest, picking_in)
-        analytic_1 = self.env['account.analytic.account'].create({
-            'name': 'Test account #1',
-        })
-        res.update({'analytic_account_id': analytic_1.id})
+        analytic_1 = cls.env["account.analytic.account"].create(
+            {"name": "Test account #1"}
+        )
+        res.update({"analytic_account_id": analytic_1.id})
         return res
 
-    def test_analytic(self):
-        for line in self.rma_ana_id.rma_line_ids:
+    def test_analytic(cls):
```

Comment:
here use `self` instead

Reply:
classmethod should work, shouldn't it?

Reply:
in the testing methods named `test_something` no, they always are with self and without classmethod

Reply:
i-dont_know_what_am_i_doing.png

Reply:
I think you could to the same in tests of other modules like rma

Reply:
Not priority IMHO, I think it is better for 13.0